### PR TITLE
Myriad 200

### PR DESCRIPTION
**Pull request by hokiegeek2**
Encompasses expanded JUnit test coverage

### CODE REVIEW

File: myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java [hunk 1]
```diff
@@ -58,6 +58,8 @@ public synchronized void initInternal(Configuration conf) throws Exception {
       //something is broken
       throw new RuntimeException("Could not find valid updateFile Method");
     }
+    
+    startInternal();
```
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Why was this added?

Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: HI Darin, I added this because the Junit test I wrote was throwing a NPE, this fixed it, and then I forgot to go back and figure out a more permanent solution that does not involve multiple invocations of startInternal.  I'll do that ASAP and check back in later today

File: myriad-scheduler/src/main/java/org/apache/myriad/Main.java [hunk 2]
```diff
@@ -76,16 +76,16 @@
...
+    new Main().run(injector.getInstance(MyriadConfiguration.class));
+  }
+  
+  private static void initConfiguration(Configuration conf, AbstractYarnScheduler scheduler, RMContext context, InterceptorRegistry registry) {
+    MyriadModule myriadModule = new MyriadModule("myriad-config-default.yml", conf, scheduler, context, registry);
     MesosModule mesosModule = new MesosModule();
     injector = Guice.createInjector(myriadModule, mesosModule, new WebAppGuiceModule());
-
-    new Main().run(cfg);
   }
```
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/Main.java]: I'm not sure what adding this method buys us in terms of readability, is it for unit testing?

File: myriad-scheduler/src/main/java/org/apache/myriad/Main.java [hunk 3]
```diff
@@ -158,10 +158,8 @@ private void initProfiles(Injector injector) {
           Long cpu = Long.parseLong(profileResourceMap.get("cpu"));
           Long mem = Long.parseLong(profileResourceMap.get("mem"));
 
-          ServiceResourceProfile serviceProfile = new ExtendedResourceProfile(new NMProfile(profile.getKey(), cpu, mem),
+          ServiceResourceProfile serviceProfile = new ExtendedResourceProfile(new NMProfile(profile.getKey(), cpu, mem), taskUtils.getExecutorCpus(), taskUtils.getExecutorMemory(),
               taskUtils.getNodeManagerCpus(), taskUtils.getNodeManagerMemory());
-          serviceProfile.setExecutorCpu(taskUtils.getExecutorCpus());
-          serviceProfile.setExecutorMemory(taskUtils.getExecutorMemory());
```
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/Main.java]: Putting a comment here the nodemanager/executor cpu memory limits a relic of how we used to run the executor which was as a small supervisor task.  We should clean this up.  We should verify the setting gets passed all the way though to the nodemanager as well.

File: myriad-scheduler/src/main/java/org/apache/myriad/MyriadModule.java [hunk 4]
```diff
@@ -168,4 +172,16 @@ ExecutorCommandLineGenerator providesCLIGenerator(MyriadConfiguration cfg) {
     }
     return cliGenerator;
   }
+  
+  protected MyriadConfiguration generateMyriadConfiguration(String configFile) {
+    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
```
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/MyriadModule.java]: I think a method like this exists in TaskUtils (which may not be the right place for it).  Does it meet your needs?

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ExtendedResourceProfile.java [hunk 5]
```diff
@@ -33,10 +33,23 @@
...
+  * @param cpu
+  * @param mem          will throw NullPoiterException if childProfile is null
+  */
   public ExtendedResourceProfile(NMProfile childProfile, Double cpu, Double mem) {
     super(childProfile.getName(), cpu, mem);
+
     this.childProfile = childProfile;
-    this.className = ExtendedResourceProfile.class.getName();
+    this.className    = ExtendedResourceProfile.class.getName();
   }
```
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ExtendedResourceProfile.java]: nit: extra whitespace

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceResourceProfile.java [hunk 6]
```diff
@@ -44,19 +44,25 @@
...
-    this.className = ServiceResourceProfile.class.getName();
   }
 
+  public ServiceResourceProfile(String name, Double cpus, Double mem, Double execCpus, Double execMemory) {
+    this.name           = name;
+    this.cpus           = cpus;
+    this.memory         = mem;
+    this.executorCpu    = execCpus;
+    this.executorMemory = execMemory;
+  }
```
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceResourceProfile.java]: White space

File: myriad-scheduler/src/test/java/org/apache/myriad/configuration/MyriadConfigurationTest.java [hunk 7]
```diff
@@ -21,71 +21,51 @@
...
-    assertEquals(false, cfg.isCheckpoint());
-    assertEquals(true, cfg.isHAEnabled());
-    assertEquals(false, cfg.isRebalancerEnabled());
   }
-  
+
   @Test
-  public void executorConfigurationTest() throws Exception {
+  public void testExecutorConfiguration() throws Exception {
     MyriadExecutorConfiguration conf = cfg.getMyriadExecutorConfiguration();
```
Comment (DarinJ) [inline myriad-scheduler/src/test/java/org/apache/myriad/configuration/MyriadConfigurationTest.java]: If we're going to change the test naming style configuration it needs to be a complete change.  Have you verified this?
@yufeldman @smarella @sdaingade @mohitsoni any strong opinions on style?

**Comment by DarinJ** [comment]
Wow that's a lot more test coverage.  There are a few areas of the code you've changed I'm unfamiliar with, so I'd appreciate some more eye's on it.  I'll go ahead and try to verify it doesn't break anything.

@hokiegeek2 have you tried verifying you can build and test against varying versions of hadoop? You can do this with:
```
./gradlew -PhadoopVer=2.7.2 build
```

File: myriad-scheduler/src/test/java/org/apache/myriad/configuration/MyriadConfigurationTest.java (hunk 7 above)
Comment (yufeldman) [inline myriad-scheduler/src/test/java/org/apache/myriad/configuration/MyriadConfigurationTest.java]: I do not think this change is necessary, as usage of annotations allows to have any naming for the tests.

File: myriad-scheduler/src/main/java/org/apache/myriad/MyriadModule.java (hunk 4 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/myriad/MyriadModule.java]: Could be, will take a look and update accordingly

**Comment by hokiegeek2** [comment]
Hi Darin,

I blipped out on testing with various versions of Hadoop, will do this weekend.

I deployed to our AWS cluster (Hadoop 2.6.0), did flex-up/down, and executed a handful of Hadoop jobs.

--John

File: myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java (hunk 1 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Okay, I've had about 5 minutes to look at this today. :) I added the @VisibleForTesting annotation to the MyriadFileSystemRMStateStore startInternal implementation and now the JUnit tests pass. Is everyone cool with this solution?

--John

Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: VisibleForTesting is fine.  Just concerned about the extra call to startInternal.

Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Cool, will update and checkin after 21 Pilots concert! :)

File: myriad-scheduler/src/main/java/org/apache/myriad/Main.java (hunk 2 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/myriad/Main.java]: No, just thought it would be cool to modularize a bit.  I can change it back if you'd prefer.

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ExtendedResourceProfile.java (hunk 5 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ExtendedResourceProfile.java]: Fixed! :)

File: myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceResourceProfile.java (hunk 6 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceResourceProfile.java]: Fixed! :)

File: myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java (hunk 1 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Done, checked in

File: myriad-scheduler/src/test/java/org/apache/myriad/configuration/MyriadConfigurationTest.java (hunk 7 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/test/java/org/apache/myriad/configuration/MyriadConfigurationTest.java]: If the consensus is to retain the test naming convention as demonstrated in the initial version of MyriadConfigurationTest, that's fine, I can make those changes.  Just need a consensus. :)

File: myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java (hunk 1 above)
Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Seems to still be there

File: myriad-scheduler/src/main/java/org/apache/myriad/MyriadModule.java (hunk 4 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/myriad/MyriadModule.java]: HI Darin,

I looked at TaskUtils and there is no method that does this. The goal of this change together with the corresponding constructor parameter list change is to consolidate Guice-based object graph population within the MyriadModule class.

Comment (DarinJ) [inline myriad-scheduler/src/main/java/org/apache/myriad/MyriadModule.java]: OK, good enough I'm trying to clean up taskutils anyway.

File: myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java (hunk 1 above)
Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Whoops, sorry about that, now it's gone

--John

Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Hold on, having a git issue

Comment (hokiegeek2) [inline myriad-scheduler/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/MyriadFileSystemRMStateStore.java]: Hi Everyone,

Addressed PR comments and did a bit more cleanup. Last unaddressed issue is
the unit test naming convention Darin mentioned. Please confirm if the
consensus is for me to change the names of my test methods to match
previously-written tests or vice versa.

Thanks

--John

File: myriad-scheduler/src/test/java/org/apache/myriad/configuration/MyriadConfigurationTest.java (hunk 7 above)
Comment (DarinJ) [inline myriad-scheduler/src/test/java/org/apache/myriad/configuration/MyriadConfigurationTest.java]: Looks like there's a mix in naming conventions already, so I don't see an issue with the rename.

**Comment by DarinJ** [comment]
OK, looks good.  Going to leave open until until 5pm PDT in case anyone else wants to chime in.
